src/taskItem.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { NodeRenderer } from '@myst-theme/providers';
+import type { ListItem } from 'myst-spec-ext';
+import { useJupyterCell } from './JupyterCellProvider';
+
+function TaskItem({
+  checked,
+  line,
+  children
+}: {
+  checked?: boolean;
+  children: React.ReactNode;
+  line?: number;
+}) {
+  // The rendering waiting on promises from Jupyter is slow
+  // By keeping state here we can render fast & optimistically
+  const [local, setLocal] = React.useState(checked ?? false);
+  const { cell } = useJupyterCell();
+  return (
+    <li className="task-list-item">
+      <input
+        type="checkbox"
+        disabled={!cell}
+        className="task-list-item-checkbox"
+        checked={local}
+        onClick={() => {
+          // Bail if no line number was found
+          if (!cell || line == null) return;
+          const text = cell.model.value.text;
+          // This is a pretty cautious replacement for the identified line
+          const lines = text.split('\n');
+          lines[line] = lines[line].replace(
+            /^(\s*(?:-|\*)\s*)(\[[\s|x]\])/,
```

Comment:
This regex is more permissive than <https://myst-tools.org/sandbox>'s parser. However, I think that's fine; I don't think someone can write MyST that produces a list element _after_ markup that matches this regex and doesn't break the remaining parse. Not super confident, though I don't think it matters; this is a WIP.